src/release_artifacts_resources/ios/cdk/cdk/credential_rotation/lambda_functions/src/handler.py

```diff
@@ -0,0 +1,113 @@
+    #           "configuration": {
+    #              "user_env_variable": "IAM_USERNAME",
+    #              "iam_role_env_variable": "IAM_ROLE"
+    #           }
+    #           "destination": {
+    #               "specifier": "aws-sdk-ios-cci",
+    #               "mapping_to_destination": [
+    #                       {
+    #                            "destination_key_name": "XCF_ACCESS_KEY_ID",
+    #                            "result_value_key": "access_key_id"
+    #                       },
+    #                       {
+    #                            "destination_key_name": "XCF_SECRET_ACCESS_KEY",
+    #                            "result_value_key": "secret_access_key"
+    #                       },
+    #                       {
+    #                            "destination_key_name": "XCF_SESSION_TOKEN",
+    #                            "result_value_key": "session_token"
+    #                       }    
+    #                   ]
+    #               }
+    #           }
+    #       },
+    #       {
+    #           "type": "secrets_manager",
+    #            "description": ""
+    #            "configuration": {
+    #                    "secret_key_env_variable": "GITHUB_CREDENTIALS_SECRET",
+    #             }
+    #            "destination": {
+    #                       "specifier": "aws-sdk-ios-cci",
+    #                       "mapping_to_destination": [
+    #                       {
+    #                            "destination_key_name": "GITHUB_SPM_TOKEN",
+    #                            "result_value_key": "GITHUB_SPM_TOKEN"
+    #                       },
+    #                       {
+    #                            "destination_key_name": "GITHUB_SPM_USER",
+    #                            "result_value_key": "GITHUB_SPM_USER"
+    #                       }
+    #                   ]
+    #               }                        
+    #       },
+    #       {
+    #           "type": "lambda_env_variables",
+    #            "description": ""
+    #            "configuration": {
+    #                    "lambda_env_var_key": "SPM_S3_BUCKET_NAME",
+    #             }
+    #            "destination": {
+    #                       "specifier": "aws-sdk-ios-cci",
+    #                       "mapping_to_destination": [
+    #                       {
+    #                            "destination_key_name": "XCF_S3_BUCKET_NAME",
+    #                       }
+    #               }                        
+    #       }
+    #   ]
+    #   "destinations": {
+    #       "aws-sdk-ios-cci": {
+    #           "type": "cci_env_variable",
+    #           "description": "Circle CI environment variable for AWS SDK iOS repo",
+    #           "github_path": "aws-amplify/aws-sdk-ios",
+    #           "secret_arn_env_variable": "CIRCLE_CI_IOS_SDK_API_TOKEN"            
+    #        }
+    #   }
+    # }
```

Comment:
JSON syntax. Please review `mapping_to_destination` -- I think we decided that was just a straight-up map, not an array of objects?:

```suggestion
    """
    Invoked with the following event structure:
    ```
    {
      "sources": [
        {
          "type": "aws_session_cred",
          "description": "Temporary AWS Credentials to upload the release artifacts to S3 and invalidate Cloudfront",
          "configuration": {
            "user_env_variable": "IAM_USERNAME",
            "iam_role_env_variable": "IAM_ROLE"
          },
          "destination": {
            "specifier": "aws-sdk-ios-cci",
            "mapping_to_destination": [
              {
                "destination_key_name": "XCF_ACCESS_KEY_ID",
                "result_value_key": "access_key_id"
              },
              {
                "destination_key_name": "XCF_SECRET_ACCESS_KEY",
                "result_value_key": "secret_access_key"
              },
              {
                "destination_key_name": "XCF_SESSION_TOKEN",
                "result_value_key": "session_token"
              }
            ]
          }
        },
        {
          "type": "secrets_manager",
          "description": "",
          "configuration": {
            "secret_key_env_variable": "GITHUB_CREDENTIALS_SECRET"
          },
          "destination": {
            "specifier": "aws-sdk-ios-cci",
            "mapping_to_destination": [
              {
                "destination_key_name": "GITHUB_SPM_TOKEN",
                "result_value_key": "GITHUB_SPM_TOKEN"
              },
              {
                "destination_key_name": "GITHUB_SPM_USER",
                "result_value_key": "GITHUB_SPM_USER"
              }
            ]
          }
        },
        {
          "type": "lambda_env_variables",
          "description": "",
          "configuration": {
            "lambda_env_var_key": "SPM_S3_BUCKET_NAME"
          },
          "destination": {
            "specifier": "aws-sdk-ios-cci",
            "mapping_to_destination": [
              {
                "destination_key_name": "XCF_S3_BUCKET_NAME"
              }
            ]
          }
        }
      ],
      "destinations": {
        "aws-sdk-ios-cci": {
          "type": "cci_env_variable",
          "description": "Circle CI environment variable for AWS SDK iOS repo",
          "github_path": "aws-amplify/aws-sdk-ios",
          "secret_arn_env_variable": "CIRCLE_CI_IOS_SDK_API_TOKEN"
        }
      }
    }
    ```
    """

```